### ISSUE TITLE
UserId is generated with curly braces in production

### ISSUE THREAD

**Issue by ChristianCRM** [kip-dk/angular-xrmservice#11]
Hi Kip,
I'm getting an error on creation of records because the id of the current logged in user returned by the function contains the curly braces. This happens only for production builds and it works fine on development.,
Could you have a look into it please?

**Comment by kip-dk** [comment]
Hi chistian are you assigning the reference field (assume owner or similar) through an entity reference or directly. Could you share a code snipper here where i can see how you assign the field in the entity, then i will take a look.   in the mean while you can simply wrap the out in your component  
 myentity.myfield.id = userid.replace('{','').replace('}','') when assigning the id.

I appriciate your input. thanks.

**Comment by kip-dk** [comment]
just for you understanding, the difference between developer mode and running inside dynamics 365, is when running inside dynamic 365, the context actually comes from CRM, while in developermode, the framework is building a fake crm-context.

**Comment by ChristianCRM** [comment]
Hi Kip,
Please see my snippet attached.
![image](https://user-images.githubusercontent.com/28629328/61520337-df55a080-aa05-11e9-986f-ce567d8f3b73.png)

I have solved the problem by using the GetOrganizationInfo() which uses the WhoAmI function to retrieve the userid. This works fine as the guid does not contain the braces.

In development the getCurrentUserId() returns the userid without the braces, but when you build you app for production it returns it with the braces. It's not a problem by any means, what I'm saying is that the behaviour is not consistent and causes unexpected result when moving from an environment to another.

Hope this helps.

**Comment by kip-dk** [comment]
Hi Christian. 

To ensure consistency, i would need to add {} in the fake context, because the impl. of context when running inside dynamics 365 as web resource is given by dynamis 365 it self, and i do not wish to wrap that cat. Could you share with me what is going on inside the ActivityParty constructor. The framework is doing a lot of work, remving {} from entity reference id's if applicable, so i would like to se witch entity property assign strategy you are using so i can see witch scenario i do not hit.   thanks.

**Comment by ChristianCRM** [comment]
Hi Kip,

This is the constructor for the activity party and I think the problem is that I'm using the Entity base class to accept any Entity type from Dynamics. I then use the id property to get the id for the activity party to be pushed to crm.

![image](https://user-images.githubusercontent.com/28629328/61616180-b892a680-ac5f-11e9-904d-2cc0d02c0373.png)

Hope this helps.

**Comment by kip-dk** [comment]
aha, ok, the first screendump was actually enough. You are creating an entity, and assign the id of the entity explicit (be aware that the ctxcontext will not know  about this instance, and if parsing it to an update method it will proberbly fail because it is not changedtracked), and then you parse that entity in as an entity reference to the ActivityParty. Looks like a logical thing to do. I will create a test case for this and make sure that the pattern works in all situations..

**Comment by ChristianCRM** [comment]
Correct, I use the base class as all the other models inherit from Entity and because Activity Party must accept a number of different entities. The only field I need is the logical name and the id as shown below:

![image](https://user-images.githubusercontent.com/28629328/61626295-d585a480-ac74-11e9-8ce4-10644c83f81a.png)
